src/Xrefcheck/System.hs

```diff
@@ -61,25 +61,20 @@ mkGlobPattern path = do
         Special characters in file names can be escaped using square brackets, e.g. <a> -> [<]a[>].
         |]
 
-normaliseGlobPattern :: RelGlobPattern -> RelGlobPattern
-normaliseGlobPattern = RelGlobPattern . normaliseWithNoTrailing . coerce
-
-bindGlobPattern :: FilePath -> RelGlobPattern -> Glob.Pattern
-bindGlobPattern root (RelGlobPattern relPat) = readingSystem $ do
-  -- TODO [#26] try to avoid using canonicalization
-  absPat <- canonicalizePath (root </> relPat)
+bindGlobPattern :: CanonicalPath -> RelGlobPattern -> Glob.Pattern
+bindGlobPattern root (UnsafeRelGlobPattern relPat) = readingSystem $ do
+  absPat <- canonicalizePath $ unCanonicalPath root </> relPat
```

Comment:
Suggestion: let's extract this appending to a separate function next to `CanonicalPath`. As well as the [substraction](https://github.com/serokell/xrefcheck/pull/230/files#diff-a2387580ec67193e7aca27b551d125c87d933dd0b499daaddaba74e1e528cd71R531).